src/Sparrow/Utils/TaskExecuter.cs

```diff
@@ -56,8 +56,8 @@ public Runner()
 
         public static void CompleteAndReplace(ref TaskCompletionSource<object> task)
         {
-            task = Interlocked.Exchange(ref task, new TaskCompletionSource<object>());
-            Execute(TaskCompletionCallback, task);
+            var task2 = Interlocked.Exchange(ref task, new TaskCompletionSource<object>());
```

Comment:
Because we are also replacing the task itself, then setting it back to the replaced value, so we didn't actually replaced it.
Note that we have ref on _both_ values.